### PR TITLE
fix(core): Filter out workflows that failed to activate on startup

### PR DESCRIPTION
**Pull request by ivov**
To test manually:
1. Set up workflow 1 with a trigger with an invalid cred.
2. Set up workflow 2 with an error trigger. Set this workflow to save executions.
3. Set workflow 1 to have workflow 2 as error workflow.
4. In DB edit `workflow_entity.active` for workflow 1 to `true`
5. Run `packages/cli/bin/n8n start --tunnel`

Expected:
- ~Workflow 1 should be deactivated and displayed as inactive on canvas.~ Workflow 1 should be shown with red activator on canvas.
- Workflow 1 should not keep retrying to activate.
- Workflow 2 should have executed.

### CODE REVIEW

**Review by krynble — CHANGES_REQUESTED** [review]
I think one of the issues that we are having is that the `packages/cli/src/ActiveWorkflowRunner.ts`'s `activationErrors` property is not getting set correctly.

We need to show the red triangle with exclamation mark properly as in the code in `packages/editor-ui/src/components/WorkflowActivator.vue` (look for `displayActivationError`)

We gotta make sure that this works properly and users can see this in the UI.

**Comment by ivov** [comment]
> I think one of the issues that we are having is that the `packages/cli/src/ActiveWorkflowRunner.ts`'s `activationErrors` property is not getting set correctly.
> 
> We need to show the red triangle with exclamation mark properly as in the code in `packages/editor-ui/src/components/WorkflowActivator.vue` (look for `displayActivationError`)
> 
> We gotta make sure that this works properly and users can see this in the UI.

Got you, I didn't know this. Please see update.

**Review by krynble — APPROVED** [review]
LGTM

This will cover only one scenario where the error message contains "Authorization" only, so maybe we still fail to detect other cases, but since we added the retry policy we can't simply fail and give up.

Maybe @netroy has some ideas on how to improve this, but it looks good in the problem it needs to address.

**Comment by ivov** [comment]
@krynble Based on our discussion, I just now tested the follow-up case:
1. Set up workflow 1 with a trigger with an invalid cred.
2. Set up workflow 2 with an error trigger. Set this workflow to save executions.
3. Set workflow 1 to have workflow 2 as error workflow.
4. In DB edit `workflow_entity.active` for workflow 1 to `true`
5. Run `packages/cli/bin/n8n start --tunnel` → Workflow 1 shown with red activator on canvas.
6. Update cred to be valid.
7. Deactivate and reactivate workflow → Workflow 1 starts listening for events.

Tested with GH trigger.

File: packages/cli/test/unit/ActiveWorkflowRunner.test.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ const generateWorkflows = (count: number): WorkflowEntity[] => {
 	for (let i = 0; i < count; i++) {
 		const workflow = new WorkflowEntity();
 		Object.assign(workflow, {
-			id: i + 1,
+			id: (i + 1).toString(),
```
Comment (netroy) [inline packages/cli/test/unit/ActiveWorkflowRunner.test.ts:54]: thanks for this fix. [I've had this change in a branch for weeks](https://github.com/n8n-io/n8n/commit/84613648334ffeef1336d932ea98df5fba852f7f), but haven't had the opportunity to sneak it in a PR 🙏🏽

**Review by netroy — APPROVED** [review]

File: packages/cli/test/unit/ActiveWorkflowRunner.test.ts (hunk 1 above)
Comment (netroy) [reply packages/cli/test/unit/ActiveWorkflowRunner.test.ts:54]: in general, I'd really like us to move away from using `Object.assign` for creating entities, and use `repository.create` instead, as that's type-safe.

**Comment by janober** [comment]
Got released with [`n8n@0.236.3`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.236.3)

**Pull request merged**